server/src/main/java/org/apache/druid/metadata/SqlSegmentsMetadataManager.java

```diff
@@ -956,12 +956,30 @@ public Optional<Iterable<DataSegment>> iterateAllUsedNonOvershadowedSegmentsForD
                    .transform(timeline -> timeline.findNonOvershadowedObjectsInInterval(interval, Partitions.ONLY_COMPLETE));
   }
 
+  /**
+   * Retrieves segments for a given datasource that are marked unused and that are *fully contained by* any interval
+   * in a particular collection of intervals. If the collection of intervals is empty, this method will retrieve all
+   * unused segments.
+   *
+   * This call does not return any information about realtime segments.
+   *
+   * @param datasource      The name of the datasource
+   * @param interval        The intervals to search over
+   * @param limit           The limit of segments to return
+   * @param offset          The offset to use when retrieving matching segments.
+   * @param orderByStartEnd Specifies the order with which to return the matching segments by start time, end time. A
```

Comment:
An enum type would be a better choice for the direction of order by? Similar to `OrderByColumnSpec#Direction`

Reply:
Thanks! Added new enum